### PR TITLE
(Ops) List2List

### PR DESCRIPTION
**Pull request by SoluMilken**
1. add_sos_eos: Insert `start token` to the head of input, then append `end token` at the end.
2. pad: Pad input sequence to a given length with `pad token`.
3. token2index: Assign token to an ordinal (integer)  based on given token2index mapping.

### CODE REVIEW

File: uttut/pipeline/edit/lst2lst.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 from .replacement import ReplacementGroup
 from .utils import (
     _transform_sequence,
     _gen_inverse_replacement_group,
 )
 
 
-def apply(input_lst: list, replacement_group: ReplacementGroup) -> List[str]:
-    output: List[list]
+def apply(input_lst: list, replacement_group: ReplacementGroup) -> list:
```
Comment (Liangtaiwan) [inline uttut/pipeline/edit/lst2lst.py]: 為毛不用typing了？

Comment (SoluMilken) [reply uttut/pipeline/edit/lst2lst.py]: 因為 
如果是 input_list 是 list[str]  -> output 揪是 list[str]
如果是 input_list 是 list[int]  -> output 揪是 list[int]

Comment (stegben) [reply uttut/pipeline/edit/lst2lst.py]: 試試： `List[Union[str, List[str]]]`

File: uttut/pipeline/edit/replacement.py [hunk 2]
```diff
@@ -62,8 +62,41 @@ def __hash__(self):
...
+            f"{self.new_value}, annotation={self.annotation})"
+        return output_repr
+
+    def __hash__(self):
+        return hash(self.__repr__())
+
+
 class ReplacementGroup(Group):
 
+    _ServiceLocator = {
```
Comment (stegben) [inline uttut/pipeline/edit/replacement.py]: 改一下命名，這不是 `ServiceLocator`

Comment (SoluMilken) [reply uttut/pipeline/edit/replacement.py]: 等等會變成只剩Replacement

File: uttut/pipeline/edit/lst2lst.py (hunk 1 above)
Comment (SoluMilken) [reply uttut/pipeline/edit/lst2lst.py]: rrrrr不是 
是這樣 input_lst: List[baba]  ===>  output: List[baba]

File: uttut/pipeline/ops/tests/test_pad.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        [1, 2, 3, 4, 5],
+        id='equal',
+    ),
+    pytest.param(
+        ['alvin'],
+        [1],
+        ['alvin', PAD_TOKEN, PAD_TOKEN, PAD_TOKEN, PAD_TOKEN],
+        [1, 0, 0, 0, 0],
+        id='shorter',
+    ),
```
[the remark this answers is not in the document]
Comment (SoluMilken) [reply uttut/pipeline/ops/tests/test_pad.py]: 在這裡 https://github.com/Yoctol/uttut/pull/53/files/d3dd790ed1eba0a93ac8c6b5f5994c8b7221d7c1#diff-783ee55e5d4f7d471516aa8a4858fc8cR38

**Pull request merged**